### PR TITLE
Fix Windows issues (WIP)

### PR DESCRIPTION
**Pull request by apedroferreira**
Closes https://github.com/mui/mui-toolpad/issues/1891

Didn't find any more issues besides:
- error in typings script when running the project in development
- unable to use/import functions (reported above)

### CODE REVIEW

File: packages/toolpad-app/src/toolpadDataSources/local/server.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ function formatCodeFrame(location: esbuild.Location): string {
 }
 
 async function createMain(): Promise<string> {
-  const relativeFunctionsFilePath = [`.`, getFunctionsFile('.')].join(path.sep);
+  const relativeFunctionsFilePath = path.resolve([`.`, getFunctionsFile('.')].join(path.sep));
```
Comment (Janpot) [inline packages/toolpad-app/src/toolpadDataSources/local/server.ts:81]: Why can't we use relative paths on windows? I intended it to be relative as to keep the file portable.

File: packages/toolpad-app/scripts/typings.mts [hunk 2]
```diff
@@ -38,7 +38,7 @@ async function main() {
       LIBS.map(async ({ name }) => {
         const files: { filename: string; moduleId: string }[] = [];
         const resolvedPkg = require.resolve(`${name}/package.json`);
-        const pkgJson = await import(resolvedPkg, { assert: { type: 'json' } });
+        const pkgJson = await import(url.pathToFileURL(resolvedPkg).toString(), { assert: { type: 'json' } });
```
Comment (Janpot) [inline packages/toolpad-app/scripts/typings.mts:41]: Not sure what benefit the `import()` brings us. I'm fine with using `readJsonFile ` as well

File: packages/toolpad-app/src/toolpadDataSources/local/server.ts (hunk 1 above)
Comment (apedroferreira) [reply packages/toolpad-app/src/toolpadDataSources/local/server.ts:81]: I just tried to fix the issues at first, didn't dig enough to understand them 100% yet or see if there's a better solution so I'll still do that.

File: packages/toolpad-app/scripts/typings.mts (hunk 2 above)
Comment (apedroferreira) [reply packages/toolpad-app/scripts/typings.mts:41]: Okay - I found similar issues and fixes to this one in other libraries, I'll probably post them here too once the PR is ready, this seems like a weird issue.

File: packages/toolpad-app/src/toolpadDataSources/local/server.ts (hunk 1 above)
Comment (Janpot) [reply packages/toolpad-app/src/toolpadDataSources/local/server.ts:81]: Oh, I think I understand now. On windows it's normalizing the path with backslashes and that makes it invalid for node.js module identifiers. I think we should do something along the lines of

```tsx
function pathToNodeImportSpecifier(importPath: string): string {
  const normalized = path.normalize(importPath).split(path.sep).join('/');
  return normalized.startsWith('/') ? normalized : `./${normalized}`;
}
```

File: packages/toolpad-app/scripts/typings.mts (hunk 2 above)
Comment (apedroferreira) [reply packages/toolpad-app/scripts/typings.mts]: This was the issue here I think: https://github.com/nodejs/node/issues/31710

File: packages/toolpad-app/src/toolpadDataSources/local/server.ts [hunk 3]
```diff
@@ -101,8 +106,8 @@ async function createMain(): Promise<string> {
     async function getResolvers() {
       if (!resolversPromise) {
         resolversPromise = (async () => {
-          const functions = await import(${JSON.stringify(
-            relativeFunctionsFilePath,
+          const functions = await import(${pathToNodeImportSpecifier(
```
Comment (Janpot) [inline packages/toolpad-app/src/toolpadDataSources/local/server.ts:109]: doesn't this have to be the other way around? json.stringify on the outside

Comment (apedroferreira) [reply packages/toolpad-app/src/toolpadDataSources/local/server.ts:109]: Maybe, it's broken on Mac now, I'm fixing it.

Comment (Janpot) [reply packages/toolpad-app/src/toolpadDataSources/local/server.ts:109]: json.stringify is there to add quotes around the string and correctly escape its content

Comment (apedroferreira) [reply packages/toolpad-app/src/toolpadDataSources/local/server.ts:109]: Yeah that was just it, I was just having some weird issues on windows VSCode.

**Review by Janpot — APPROVED** [review]
👌

**Pull request merged**